package-system/astc-encoder/Findastc-encoder.cmake.template

```diff
@@ -0,0 +1,34 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project. For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+
+# this file actually ingests the library and defines targets.
+
+
+set(LIB_NAME "astc-encoder")
+set(TARGET_WITH_NAMESPACE "3rdParty::$${LIB_NAME}")
+
+set($${LIB_NAME}_INCLUDE_DIR $${CMAKE_CURRENT_LIST_DIR}/$${LIB_NAME}/include)
+set($${LIB_NAME}_LIBRARY_DIR $${CMAKE_CURRENT_LIST_DIR}/$${LIB_NAME}/bin)
+
+if ($${PAL_PLATFORM_NAME} STREQUAL "Windows")
+    set($${LIB_NAME}_LIBRARY_DIR_WINDOWS $${$${LIB_NAME}_LIBRARY_DIR}/$$<IF:$$<CONFIG:Debug>,Debug,Release>)
+    set($${LIB_NAME}_LIBRARY   $${$${LIB_NAME}_LIBRARY_DIR_WINDOWS}/astcenc-sse2-static.lib)
+elseif ($${PAL_PLATFORM_NAME} STREQUAL "Linux")
+    set($${LIB_NAME}_LIBRARY   $${$${LIB_NAME}_LIBRARY_DIR}/libastcenc-sse2-static.a)
+else()
+    set($${LIB_NAME}_LIBRARY   $${$${LIB_NAME}_LIBRARY_DIR}/libastcenc-native-static.a)
+endif()
+
+# declare the target so that others can 'depend on it'
+add_library($${TARGET_WITH_NAMESPACE} INTERFACE IMPORTED GLOBAL)
+
+# add include directory
+ly_target_include_system_directories(TARGET $${TARGET_WITH_NAMESPACE} INTERFACE $${$${LIB_NAME}_INCLUDE_DIR})
+
+# add static library for linking
+target_link_libraries($${TARGET_WITH_NAMESPACE} INTERFACE $${$${LIB_NAME}_LIBRARY})
```

Comment:
So there is a new way to define the STATIC library dependencies for a 3rdParty library, that involes making a `STATIC IMPORTED GLOBAL` target which can be used to associate the static library with the `3rdParty::astc-encoder` target, instead of having the `3rdParty::astc-encoder` target depend on path to the static library.

This involves setting the IMPORTED_LOCATION property for the library target
```suggestion
if ($${PAL_PLATFORM_NAME} STREQUAL "Windows")
    set($${LIB_NAME}_LIBRARY_RELEASE $${$${LIB_NAME}_LIBRARY_DIR}/Releaseastcenc-sse2-static.lib)
    set($${LIB_NAME}_LIBRARY_DEBUG $${$${LIB_NAME}_LIBRARY_DIR}/Debug/astcenc-sse2-static.lib
elseif ($${PAL_PLATFORM_NAME} STREQUAL "Linux")
    set($${LIB_NAME}_LIBRARY_RELEASE $${$${LIB_NAME}_LIBRARY_DIR}/libastcenc-sse2-static.a)
    set($${LIB_NAME}_LIBRARY_DEBUG $${$${LIB_NAME}_LIBRARY_RELEASE})
else()
    set($${LIB_NAME}_LIBRARY_RELEASE $${$${LIB_NAME}_LIBRARY_DIR}/libastcenc-native-static.a)
    set($${LIB_NAME}_LIBRARY_DEBUG $${$${LIB_NAME}_LIBRARY_RELEASE})
endif()

# declare the target so that others can 'depend on it'
add_library($${TARGET_WITH_NAMESPACE} INTERFACE IMPORTED GLOBAL)

# add include directory
ly_target_include_system_directories(TARGET $${TARGET_WITH_NAMESPACE} INTERFACE $${$${LIB_NAME}_INCLUDE_DIR})

# add static library as "output" IMPORTED_LOCATION for the target
set_target_properties($${TARGET_WITH_NAMESPACE}
    PROPERTIES
        IMPORTED_LOCATION_DEBUG $${$${LIB_NAME}_LIBRARY_DEBUG}
        IMPORTED_LOCATION $${$${LIB_NAME}_LIBRARY_RELEASE}
)
```

Reply:
It's good to know. 
I would prefer to keep it as it is since this change would effect all the packages which requires too many follow up validation which I don't think worth it. 
How do you think?

Reply:
Well if we don't ever do this in any PRs of converting a 3rdParty library now, then no one would ever do it.

Quick question for @VickyAtAZ:
What was your process for adding the ASTCEncoder `Findastc-encoder.template.cmake` script?
Did you take another 3p package .cmake script and adapt that script to work with the ASTCEncoder?
Or did you use the instructions in the 3rdParty runbook?

I am just trying to get a feel for the process that developers are using to convert 3rdParty packages.

Reply:
I copied one from another script (squish-ccr in this case which I created few weeks ago ). 
I was looking at the run book when I was creating squish-ccr script. And the change about set_target_properties was not there yet.

Reply:
> 
> 
> Well if we don't ever do this in any PRs of converting a 3rdParty library now, then no one would ever do it.
> 
Unless the majority of the existing scripts are converted, this won't be meaningful. 
Are we going to ask engineers to convert each script individually? I don't think it's the best of using engineers' time. 
If this API is really matters, then we should have a batch converting instead one by one. 

Anyway, I don't think this is necessary for this one script especially it doesn't have any risk of compilation order issue.

Reply:
I will update the script.